src/connector/src/sink/iceberg/mod.rs

```diff
@@ -311,9 +322,53 @@ impl IcebergSink {
                 }
             };
 
+            let partition_fields = match &self.config.partition_by {
+                Some(partition_by) => {
+                    let mut partition_fields =
+                        Vec::<UnboundPartitionField>::with_capacity(partition_by.len());
+                    for partition_field in partition_by {
+                        let re = Regex::new(r"\w+(\(\w+\))?").unwrap();
```

Comment:
This Regex can be created outside the for loop. Also according to `v1, identity(v1), truncate(4,v2), bucket(5,v1), year(v3), month(v4), day(v5), hour(v6), void(v1)`, maybe we can check the prefix first and capture the param according to different transform type